### PR TITLE
remove --api-version

### PR DESCRIPTION
**Pull request by xilabao**

### CODE REVIEW

**Comment by sttts** [comment]
@k8s-bot ok to test

**Comment by sttts** [comment]
Shouldn't we remove this flag? Since when is it deprecated? Since 1.2?

/cc @kubernetes/sig-cli-api-reviews

**Comment by fabianofranz** [comment]
I'd say the flag can be completely removed (not just mark hidden). cc @deads2k

File: pkg/kubectl/cmd/explain.go [hunk 1]
```diff
@@ -57,7 +57,6 @@ func NewCmdExplain(f cmdutil.Factory, out, cmdErr io.Writer) *cobra.Command {
 		},
 	}
 	cmd.Flags().Bool("recursive", false, "Print the fields of fields (Currently only 1 level deep)")
-	cmd.Flags().String("api-version", "", "The API version to use when talking to the server")
```
Comment (deads2k) [inline pkg/kubectl/cmd/explain.go]: I think you still need this one to deal with explaining resources for particular versions.  Is there another way to describe those?

Comment (xilabao) [inline pkg/kubectl/cmd/explain.go]: Do you mean this? PTAL @deads2k

**Comment by xilabao** [comment]
Can someone approve this pr? thx. @sttts @fabianofranz @jbeda @kubernetes/sig-cli-api-reviews

**Comment by sttts** [comment]
@eparis approved?

**Pull request merged**